### PR TITLE
[CHK-1481] fix: update carts amount

### PR DESCRIPTION
**Pull request by MarcoAGreco**
#### List of Changes

- Update cart notices amount after activation

#### Motivation and Context

- Avoid amount inconsistency between getCarts and payment activation
#### How Has This Been Tested?
- build, unit

#### Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/utils/api/helper.ts [hunk 1]
```diff
@@ -262,6 +262,29 @@ const activePaymentTask = (
...
+                const cartInfo = getSessionItem(SessionItems.cart) as
+                  | Cart
+                  | undefined;
+                if (cartInfo !== undefined) {
+                  const updatedPaymentNotices = cartInfo.paymentNotices.map(
+                    (paymentNotice) => {
+                      const updatedAmount = responseType.value.payments.find(
+                        (p) =>
+                          p.rptId ===
+                          (`${paymentNotice.fiscalCode}${paymentNotice.noticeNumber}` as RptId)
```
Comment (giovanniberti) [inline src/utils/api/helper.ts:274]: Is this cast ok? Isn't `RptId` an `io-ts` tagged type?

File: src/utils/api/helper.ts [hunk 2]
```diff
@@ -262,6 +262,29 @@ const activePaymentTask = (
...
               if (responseType.status === 200) {
                 reason = "";
+                const cartInfo = getSessionItem(SessionItems.cart) as
+                  | Cart
+                  | undefined;
+                if (cartInfo !== undefined) {
+                  const updatedPaymentNotices = cartInfo.paymentNotices.map(
+                    (paymentNotice) => {
+                      const updatedAmount = responseType.value.payments.find(
+                        (p) =>
```
Comment (giovanniberti) [inline src/utils/api/helper.ts:272]: [**suggestion (if-minor)**:](https://conventionalcomments.org/) _map_ + _find_ makes this an $O(n^2)$ algorithm, what if we first map the actualized amounts from the activate response into a `RptId -> amount` map and then update the amounts?

Comment (MarcoAGreco) [reply src/utils/api/helper.ts]: You're right! Solved with 3c936a5 😁

File: src/utils/api/helper.ts (hunk 1 above)
Comment (MarcoAGreco) [reply src/utils/api/helper.ts]: I hope it is, we use it several times in this PR

Comment (giovanniberti) [reply src/utils/api/helper.ts]: Just checked, _as a type_ `RptId` is an alias of `string & PatternedString<Pattern>`, so can be safely coerced with `as`

**Review by giovanniberti — APPROVED** [review]
🚀

**Pull request merged**